gcp/appengine/blog/content/posts/using-the-determineversion-api.md

```diff
@@ -0,0 +1,184 @@
+        ```
+        And we get a response:
+
+            ```json
+            {
+            "vulns": [
+                {
+                "id": "OSV-2021-777",
+                "summary": "Heap-use-after-free in xmlAddNextSibling",
+                "details": "OSS-Fuzz report: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=34461\n\n```\nCrash type: Heap-use-after-free READ 4\nCrash state:\nxmlAddNextSibling\nxmlXIncludeCopyRange\nxmlXIncludeCopyXPointer\n```\n",
+                "modified": "2023-05-19T14:06:37.864410Z",
+                "published": "2021-05-20T00:00:30.166614Z",
+                "references": [
+                    {
+                    "type": "REPORT",
+                    "url": "https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=34461"
+                    }
+                ],
+                "affected": [
+                    {
+                    "package": {
+                        "name": "libxml2",
+                        "ecosystem": "OSS-Fuzz",
+                        "purl": "pkg:generic/libxml2"
+                    },
+                    "ranges": [
+                        {
+                        "type": "GIT",
+                        "repo": "https://gitlab.gnome.org/GNOME/libxml2.git",
+                        "events": [
+                            {
+                            "introduced": "6c128fd58a0e4641c23a345d413672494622db1b"
+                            }
+                        ]
+                        }
+                    ],
+                    "versions": [
+                        "CVE-2021-3541",
+                        "v2.9.11",
+                        "v2.9.12",
+                        "v2.9.13",
+                        "v2.9.14",
+                        "v2.10.0",
+                        "v2.10.1",
+                        "v2.10.2",
+                        "v2.10.3",
+                        "v2.10.4",
+                        "v2.11.0",
+                        "v2.11.1",
+                        "v2.11.2",
+                        "v2.11.3",
+                        "v2.11.4"
+                    ],
+                    "ecosystem_specific": {
+                        "severity": "HIGH"
+                    },
+                    "database_specific": {
+                        "source": "https://github.com/google/oss-fuzz-vulns/blob/main/vulns/libxml2/OSV-2021-777.yaml"
+                    }
+                    }
+                ],
+                "schema_version": "1.4.0"
+                }
+            ]
+            }
+            ```
+5. Finally, we consider the response and draw conclusions. 
+
+Our copy of libxml2 returned a number of versions with similar confidence scores. 
+- Version 2.11.3, 0.718
+- Version 2.11.4, 0.718
+- Version 2.11.2, 0.707
+- Version 2.11.0, 0.697
+
+To be sure we have caught any potential vulnerabilities, we could make further queries for these other versions and see if any other vulnerabilities are found. 
```

Comment:
I would add a note here saying that it is very likely the version/s that have the highest confidence score is very likely the closest version to the actual code, more than the small difference in the score indicates. E.g. even though there's only a 0.011 difference in score between 2.11.2 and 2.11.4, it is overwhelmingly more likely (95%+) that 2.11.4 is going to be the closer version to the actual code)

To put it another way, the score is more of how similar the submitted code is to the version, and less a probability of what version the submitted code is. So while it's safer to also make further queries against the other versions, most of the time it's fine to just rely on the top result that's returned. 

Not sure how to word this in a concise and understandable way.